WooCommerce/UITestsFoundation/Screens/MyStore/MyStoreScreen.swift

```diff
@@ -27,4 +27,46 @@ public final class MyStoreScreen: ScreenObject {
         topBannerCloseButton.tap()
         return self
     }
+
+    func tapTimeframeTab(timeframe: String) -> MyStoreScreen {
+        app.cells.staticTexts[timeframe].tap()
+
+        return self
+    }
+
+    public func goToThisWeekTab() -> MyStoreScreen {
+        return tapTimeframeTab(timeframe: "This Week")
+    }
+
+    public func goToThisMonthTab() -> MyStoreScreen {
+        return tapTimeframeTab(timeframe: "This Month")
+    }
+
+    public func goToThisYearTab() -> MyStoreScreen {
+        return tapTimeframeTab(timeframe: "This Year")
+    }
+
+    func verifyStatsForTimeframeLoaded(timeframe: String) -> MyStoreScreen {
+        let textPredicate = NSPredicate(format: "label MATCHES %@", "Store revenue chart \(timeframe)")
+        XCTAssertTrue(app.images.containing(textPredicate).element.exists, "\(timeframe) chart not displayed")
+
+        return self
+    }
```

Comment:
WDYT of extending this asserting function with one more check: making sure that the tab of interest is selected? Example of `This Year` tab being selected:

![SCR-20221205-lqg](https://user-images.githubusercontent.com/73365754/205651037-badaf3fa-7855-4cfb-818e-abb372bd23a4.png)

(For this you'll also need to provide the second argument with cell ID):

```suggestion
    func verifyStatsForTimeframeLoaded(timeframe: String, timeframeID: String) -> MyStoreScreen {
        let textPredicate = NSPredicate(format: "label MATCHES %@", "Store revenue chart \(timeframe)")
        XCTAssertTrue(app.images.containing(textPredicate).element.exists, "\(timeframe) chart not displayed")
        XCTAssertTrue(app.cells[timeframeID].isSelected, "\(timeframeID) cell not selected")
        
        return self
    }
```

Reply:
i'm leaving this as is, for now. thinking of a scenario where the tab is not selected (e.g. if tap didn't work), the image wouldn't change too so i think that extra check isn't necessary for this test to work

Reply:
Totally 👍 to leave it with no changes, just wanted to extend my thoughts about the reason for proposal: as you say, the test will still fail if tab selection did not work, because of incorrect chart name. My proposal was more about increasing the test coverage (e.g. checking for one field in the person details page VS checking for all fields). Also, it would cover a totally made up scenario, when tab switch animation did not work, but the screen itself is updated with correct chart and data. (just my thoughts behind it, let's leave it as it is now).